axelor-studio/src/main/java/com/axelor/studio/service/StudioMetaService.java

```diff
@@ -189,15 +192,18 @@ public String updateAction(String oldAction, String newAction, boolean remove) {
   }
 
   public MetaMenu createMenu(MenuBuilder builder) {
-
-    String xmlId = XML_ID_PREFIX + builder.getName();
+    String builderName = builder.getName();
+    String xmlId =
+        (builderName.startsWith(XML_ID_PREFIX)) ? builderName : XML_ID_PREFIX + builderName;
     MetaMenu menu = metaMenuRepo.findByID(xmlId);
 
     if (menu == null) {
-      menu = new MetaMenu(builder.getName());
+      menu = new MetaMenu(xmlId);
```

Comment:
Menu override will be not possible if name is different.

Reply:
Fixed